Fix crash reporter issues

Angel-Padilla opened the pull request:
- give execute permissions to crash reporter binary when constructing the FS, as the file in the machine does not have
- use the Docker image defined in the `systemd-crash-reporter` repo to build the binary for an older GLIBC version
- add `--crash` option help
- Use a template for the `crash-reporter` service
    As we want to be able to report all crashes to sentry, we need to reliably get all
the information passed to the service when instantiated on another service failure, and
according to the [systemd doc](https://www.freedesktop.org/software/systemd/man/252/systemd.exec.html):
    >"... when multiple services trigger the same unit, those variables will be not be passed. Consider using a template handler unit for that case instead: "OnFailure=handler@%n.service" for non-templated units,or "OnFailure=handler@%p-%i.service" for templated units." 

    So, the old service is replaced with a template to be reliable

- chore: move service restart logic

    When a service fails, it does not immediately calls the onFailure service dependency, as it needs to enter the fail state and stay there, when the service has the Restart= config set to restart the service when it fails, the service does not stay on the failed state but immediatlly goes to restarting skipping the call to onFailure. The only way to trigger onFailure in this case is to deplete the number of allowed restarts, and only then, onFailure is triggered

    This PR removes the ability to restart the services that have a onFailure dependency linked to `crash-reporter@.service`, allowing the onFailure to trigger every time it fails, and moves the logic to the `crash-reporter@.service` script that decides which services are expected to be restarted when they fail

This commit requires approval of the PR: [feat: Build of the crash reporter is now don in a docker container to ensure compatibility with met system #1](https://github.com/MeticulousHome/systemd-crash-reporter/pull/1) of the `systemd-crash-reporter` repo

- chore: move scripts to `/usr/local/bin`

    Scripts used to handle media insertion and crash reporting are moved that
were present in `/etc/path/to/exec` are moved to `/usr/local/bin/path/to/exec`

Angel-Padilla committed:
give execute permissions to crash reporter binary when constructing the FS

Angel-Padilla committed:
use the Docker image defined in the systemd-crash-reporter repo to build the binary for an older GLIBC

Angel-Padilla committed:
feat: add --crash option to show help function

Mimoja commented on build-components.sh:
```diff
@@ -230,7 +230,10 @@ function build_crash_reporter() {
+        #build the Docker image
+        docker build -t build-reporter .
+        #run the container to build the binary
+        sudo docker run --rm -v $(pwd):/systemd-crash-reporter build-reporter
```
In theory we don't even need to pushd into that folder anymore now

Angel-Padilla committed:
https://www.freedesktop.org/software/systemd/man/252/systemd.exec.htmlfeat:
feat: Use a template for the crash-reporter service

As we want to be able to report all crashes to sentry, we need to reliably get all
the information passed to the service when instantiated on another service failure, and
according to the [systemd doc](https://www.freedesktop.org/software/systemd/man/252/systemd.exec.html)
the environment variables used in the binary:

"... when multiple services trigger the same unit, those variables will be not be passed.
Consider using a template handler unit for that case instead: "OnFailure=handler@%n.service" for non-templated units,
or "OnFailure=handler@%p-%i.service" for templated units."

So, the old service is replaced with a template to be reliable

Angel-Padilla committed:
chore: move service restart logic

When a service fails, it does not immediately calls the onFailure service
dependency, as it needs to enter the fail state and stay there, when the
service has the Restart= config set to restart the service when it fails,
the service does not stay on the failed state but immediatly goes to restarting
skipping the call to onFailure. The only way to trigger onFAilure in this case is
to deplete the number of allowed restarts, and only then, onFailure is triggered

This commit removes the ability to restart the services that have a onFailure
dependency to restart the service when it fails, allowing the onFailure trigger
every time it fails, and moves the logic to the crash-reporter@.service script
that decides which services are expected to be restarted when they fail

Angel-Padilla committed:
move location of crash-reporter.sh

Angel-Padilla committed:
chore: remove unnecesary directory change

Angel-Padilla committed:
fix: quote '$MONITOR_UNIT'

Mimoja commented on config/etc/crash-reporter/crash-reporter.sh:
```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/bash
```
Systemd will automatically restart crashing units but we can do it for good measure

Angel-Padilla replied:
Yes, but we don't want it to do it, or it won't trigger the OnFailure service

Mimoja commented on config/etc/systemd/system/crash-reporter@.service:
```diff
@@ -4,4 +4,4 @@ After=network.target
 Type=oneshot
-ExecStart=/opt/meticulous-crash-reporter
\ No newline at end of file
+ExecStart=/etc/crash-reporter/crash-reporter.sh
```
Etc is for configs, not for binaries. Use /usr/bin of similar

Mimoja commented on system-services/rauc-hawkbit-updater.service:
```diff
@@ -15,7 +15,8 @@ ExecStartPre=+/usr/bin/bash /etc/hawkbit/create_config.sh
 TimeoutSec=60s
 WatchdogSec=5m
-Restart=on-failure
+# explicit Restart=no as previously it was set to on-failure, case that is considered in the crash-reporter
```
But why?

Angel-Padilla replied:
Why are we restraining their auto restarting options? 

Because every service that auto restarts when failed will not trigger the OnFailure services until it uses all of its auto restart tries and it won't restart anymore,  so we would only get that last failure reported, but if that is what we actually want, I will roll this back

Angel-Padilla committed:
chore: move scripts to /usr/local/bin

Scripts used to handle media insertion and crash reporting are moved that
were present in /etc/path/to/exec are moved to /usr/local/bin/path/to/exec
